python/ray/tune/trial.py

```diff
@@ -497,6 +513,9 @@ def update_last_result(self, result, terminate=False):
         self.last_update_time = time.time()
         self.result_logger.on_result(self.last_result)
 
+        if self.keep_best_checkpoint:
+            self.results_since_checkpoint.append(result["episode_reward_mean"])
```

Comment:
instead of keeping all the rewards (which might be much of them especially if checkpoint_freq is big) keep only the sum and the count. Also, make sure that you don't incremen the count if episode_reward_mean does not exist or is a NaN. And since it is small overhead, do it always, no need for if.